packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -635,15 +635,13 @@ class TextPainter {
 
   // Get the Rect of the cursor (in logical pixels) based off the near edge
   // of the character upstream from the given string offset.
-  // TODO(garyq): Use actual extended grapheme cluster length instead of
-  // an increasing cluster length amount to achieve deterministic performance.
```

Comment:
It's not possible to improve performance using the characters package as expected.  It can find characters by iterating from the start or end of a string, but at some offset into a string, it doesn't know where the character boundaries are.  Performance is likely to be worse iterating from the start/end of a long string rather than using the existing logarithmic algorithm.